prusti-common/src/vir/vir_macro.rs

```diff
@@ -0,0 +1,125 @@
+macro_rules! vir_type {
+    (Int) => {crate::vir::Type::Int};
+    (Bool) => {crate::vir::Type::Bool};
+}
+
+macro_rules! vir_local {
+    ($name: ident : $type: ident) => {
+        crate::vir::LocalVar {
+            name: stringify!($name).to_string(),
+            typ: vir_type!($type)
+        }
+    }
+}
+
+
+macro_rules! vir {
+    (# $comment: expr) => {
+        crate::vir::Stmt::Comment($comment.to_string())
+    };
+    (label $label: expr) => {
+        crate::vir::Stmt::Label($label.to_string())
+    };
+    (assert $exp: tt) => {
+        crate::vir::Stmt::Assert(
+            vir!($exp),
+            crate::vir::FoldingBehaviour::Expr,
+            crate::vir::Position::default())
+    };
+    (inhale $exp: tt) => {
+        crate::vir::Stmt::Inhale(
+            vir!($exp),
+            crate::vir::FoldingBehaviour::Expr)
+    };
+    (apply $exp: tt) => {
+        crate::vir::Stmt::ApplyMagicWand(
+            vir!($exp),
+            crate::vir::Position::default())
+    };
+    (obtain $exp: tt) => {
+        crate::vir::Stmt::Obtain(
+            vir!($exp),
+            crate::vir::Position::default())
+    };
+    (if ($exp: tt) { $($then: tt);* } else { $($elze: tt);* }) => {
+        crate::vir::Stmt::If(vir!($exp), vec![$(vir!($then)),*], vec![$(vir!($elze)),*])
+    };
+    (if ($exp: tt) { $($then: tt);* }) => {
+        crate::vir::Stmt::If(vir!($exp), vec![$(vir!($then)),*], vec![])
+    };
+    (true) => {
+        crate::vir::Expr::Const(
+            crate::vir::Const::Bool(true),
+            crate::vir::Position::default())
+    };
+    (false) => {
+        crate::vir::Expr::Const(
+            crate::vir::Const::Bool(false),
+            crate::vir::Position::default())
+    };
+    ($lhs: tt == $rhs: tt) => {
+        crate::vir::Expr::BinOp(vir::BinOpKind::Equals,
+            Box::new(vir!($lhs)),
+            Box::new(vir!($rhs)),
+            crate::vir::Position::default())
+    };
+    ($head: tt && $tail: tt) => {
+        crate::vir::Expr::BinOp(vir::BinOpKind::And,
+            Box::new(vir!($head)),
+            Box::new(vir!($tail)),
+            crate::vir::Position::default())
+    };
+    ($head: tt || $tail: tt) => {
+        crate::vir::Expr::BinOp(vir::BinOpKind::Or,
+            Box::new(vir!($head)),
+            Box::new(vir!($tail)),
+            crate::vir::Position::default())
```

Comment:
I think there are functions defined in `Expr` that build particular binary operations with a default position, would make this a bit shorter. There might be similar for `Stmt`s too?

Reply:
some `Stmt`s have, some don't .. i don't see why, exactly. tidied up a bit.